listener/avalon_listener/tcs_work_order_handler.py

```diff
@@ -99,42 +113,52 @@ def WorkOrderGetResult(self, **params):
               as defined in EEA spec 6.1.4
         Returns jrpc response as defined in EEA spec 6.1.2
         """
-        wo_id = params["workOrderId"]
-        if not is_valid_hex_str(wo_id):
-            logging.error("Invalid work order Id")
-            raise JSONRPCDispatchException(
-                JsonRpcErrorCode.INVALID_PARAMETER,
-                "Invalid work order Id"
-            )
-
-        # Work order is processed if it is in wo-response table
-        value = self.kv_helper.get("wo-responses", wo_id)
-        if value:
-            response = json.loads(value)
-            if 'result' in response:
-                return response['result']
-
-            # response without a result should have an error
-            err_code = response["error"]["code"]
-            err_msg = response["error"]["message"]
-            if err_code == EnclaveError.ENCLAVE_ERR_VALUE:
-                err_code = WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE
-            elif err_code == EnclaveError.ENCLAVE_ERR_UNKNOWN:
-                err_code = WorkOrderStatus.UNKNOWN_ERROR
-            else:
-                err_code = WorkOrderStatus.FAILED
-            raise JSONRPCDispatchException(err_code, err_msg)
+        if "workOrderId" in params:
+            wo_id = params["workOrderId"]
+            if not is_valid_hex_str(wo_id):
+                logging.error("Invalid work order Id")
+                raise JSONRPCDispatchException(
+                    JsonRpcErrorCode.INVALID_PARAMETER,
+                    "Invalid work order Id"
+                )
+
+            # Work order is processed if it is in wo-response table
+            value = self.kv_helper.get("wo-responses", wo_id)
+            if value:
+                response = json.loads(value)
+                if 'result' in response:
+                    return response['result']
+
+                # response without a result should have an error
+                err_code = response["error"]["code"]
+                err_msg = response["error"]["message"]
+                if err_msg == self.INVALID_WORKLOAD_MESSAGE:
+                    err_msg = 'Invalid workload id'
```

Comment:
Better to check the err_code and then set the required err_msg.
```
 if err_code == xxx:
     err_msg = 'Invalid workload id'
```

Reply:
We won't get the error code from enclave manager for this error.

Reply:
Thought so. @manju956 Can we return a new error code from enclave manager for this scenario. Please suggest where to where to make this change.

Reply:
Enclave manager don't add any error code as such. New error codes can be added in tcf_error.h and we need to define a class for the new error code in error.h.
Also, we need to update the enclave code to return new error code when invalid workload_id has been identified by the enclave.

Reply:
Yeah. We need to do define new error codes corresponding to workload id and enclave manager should return that. This should be checked by listener